kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/DeltaLogFile.java

```diff
@@ -34,16 +34,17 @@ public enum LogType {
   }
 
   public static DeltaLogFile forCommitOrCheckpoint(FileStatus file) {
+    String filePath = file.getPath();
     String fileName = new Path(file.getPath()).getName();
-    LogType logType = null;
+    LogType logType;
     long version = -1;
-    if (FileNames.isCommitFile(fileName)) {
+    if (FileNames.isCommitFile(filePath)) {
```

Comment:
create https://github.com/delta-io/delta/issues/3586 to unify these utility APIs